### PR TITLE
{Release} Upgrade to Azure CLI 2.4.0

### PR DESCRIPTION
**Pull request by azclibot**
Updated Azure CLI version.

### CODE REVIEW

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,119 @@
...
+* Update containerservice package (#12964)
+* Add node public IP support in CLI extension. (#13015)
+* Fixing typo in the help command (#13055)
+
+**AppConfig**
+
+* Resolve key vault reference for kv list and export commands (#12893)
+
+**Appconfig**
+
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:27]:
```suggestion
```

File: src/azure-cli-core/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,14 @@
 Release History
 ===============
 
+2.4.0
+++++++
+* Upgrade msrestazure to 0.6.3 (#12839)
+* Output error detail of exception (#12945)
+* Support track1 and track2 mgmt SDK side by side in CLI (#12952)
```
Comment (jiasli) [inline src/azure-cli-core/HISTORY.rst:10]:
```suggestion
* Support track1 and track2 mgmt SDK side by side (#12952)
```

File: src/azure-cli-core/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,14 @@
...
+* Fix generic remove property NoneType exception (#12950)
+* `az --version`: Show command instruction and detailed instruction link when updates available (#12981)
```
Comment (haroldrandom) [inline src/azure-cli-core/HISTORY.rst:12]: * Fix bug in core test test_parser that will affect exception stack while existing if argument parse raise Exception
* Make CLI respect file logging configuration option: enable_log_file and AZURE_LOGGING_ENABLE_LOG_FILE env variable
* hotfix the name of extension of dev type is None if it has incompatibility problem with core

File: src/azure-cli-core/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,14 @@
 Release History
 ===============
 
+2.4.0
+++++++
+* Upgrade msrestazure to 0.6.3 (#12839)
+* Output error detail of exception (#12945)
```
Comment (jiasli) [inline src/azure-cli-core/HISTORY.rst:9]:
```suggestion
* Output more error details of exceptions if possible (#12945)
```

File: src/azure-cli/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,119 @@
...
+* Bug fix for list key values (#12926)
+
+**AppService**
+
+* `az functionapp create`: Changed the way linuxFxVersion was being set for dotnet linux function apps. This should fix a bug that was preventing dotnet linux consumption apps from being created. (#12817)
+* ([BREAKING CHANGE]) (az webapp create:) fix to keep existing AppSettings with az webapp create. (#12865)
+* ([BREAKING CHANGE]) (az webapp up:) fix to create RG for az webapp up command when using -g flag. (#12865)
+* ([BREAKING CHANGE]) (az webapp config:) fix to show values for non-JSON output with az webapp config connection-string list (#12865)
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:35]:
```suggestion
* ([BREAKING CHANGE]) (az webapp create:) fix to keep existing AppSettings with az webapp create. (#12865)
* [BREAKING CHANGE] az webapp up: fix to create RG for az webapp up command when using -g flag. (#12865)
* [BREAKING CHANGE] az webapp config: fix to show values for non-JSON output with az webapp config connection-string list (#12865)
```

Comment (qwordy) [inline src/azure-cli/HISTORY.rst:35]:
```suggestion
* [BREAKING CHANGE] `az webapp create`: fix to keep existing AppSettings with az webapp create. (#12865)
* [BREAKING CHANGE] `az webapp up`: fix to create RG for az webapp up command when using -g flag. (#12865)
* [BREAKING CHANGE] `az webapp config`: fix to show values for non-JSON output with az webapp config connection-string list (#12865)
```

File: src/azure-cli/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,119 @@
...
+
+* Update Batch API (#12813)
+
+**Compute**
+
+* `az sig image-version create`: Add storage account type Premium_LRS (#12919)
+* `az vmss update`: Fix terminate notification update issue (#12948)
+* `az vm/vmss create`: Add support for specialized image version (#12997)
+* SIG API Version 2019-12-01. (#12899)
+* `az sig image-version create`: Add --target-region-encryption. (#12899)
```
Comment (haroldrandom) [inline src/azure-cli/HISTORY.rst:61]: append:
* Fix tests fail when running in serial due to keyvault name is duplicated in global in-momery cache

File: src/azure-cli/HISTORY.rst [hunk 7]
```diff
@@ -3,6 +3,119 @@
...
+
+**Monitor**
+
+* Support private link scenario for monitor (#12931)
+
+**Network**
+
+* Deprecate sku for public ip update command (#12898)
+* `az network private-endpoint`: Support private dns zone group. (#13038)
+* Enable local context feature for vnet/subnet parameter (#13059)
```
Comment (haroldrandom) [inline src/azure-cli/HISTORY.rst:80]: Need to append:
* fix wrong usage example in test_nw_flow_log_delete

File: src/azure-cli/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,119 @@
...
+**RDBMS**
+
+* Add Azure active directory administrator commands for PostgreSQL and MySQL (#12812)
+
+**Service Fabric**
+
+* Fix #12891: az sf application update --application-parameters removes old parameters that are not in the request (#12992)
+
+**ServiceFabric**
+
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:99]:
```suggestion
```

File: src/azure-cli/HISTORY.rst [hunk 9]
```diff
@@ -3,6 +3,119 @@
...
+* `az acr run --cmd`: disable working directory override (#12877)
+* Dedicated data endpoint (#12967)
+
+**AKS**
+
+* az aks list -o table should show privateFqdn as fqdn for private clusters (#12784)
+* Add --uptime-sla (#12772)
+* Update containerservice package (#12964)
+* Add node public IP support in CLI extension. (#13015)
+* Fixing typo in the help command (#13055)
```
Comment (qianwens) [inline src/azure-cli/HISTORY.rst:20]:
```suggestion
* Fix typo in the help command (#13055)
```

File: src/azure-cli/HISTORY.rst [hunk 10]
```diff
@@ -3,6 +3,119 @@
...
+
+**IoT Central**
+
+* Deprecate `az iotcentral` (#12681)
+* Add `az iot central` command module (#12681)
+
+**Monitor**
+
+* Support private link scenario for monitor (#12931)
+
```
Comment (haroldrandom) [inline src/azure-cli/HISTORY.rst:75]:
```suggestion
* fix wrong mocking way in test_monitor_general_operations.py

```

File: src/azure-cli/HISTORY.rst [hunk 11]
```diff
@@ -3,6 +3,119 @@
...
+++++++
+
+**ACR**
+
+* `az acr run --cmd`: disable working directory override (#12877)
+* Dedicated data endpoint (#12967)
+
+**AKS**
+
+* az aks list -o table should show privateFqdn as fqdn for private clusters (#12784)
```
Comment (qianwens) [inline src/azure-cli/HISTORY.rst:16]:
```suggestion
* `az aks list -o table` should show privateFqdn as fqdn for private clusters (#12784)
```

File: src/azure-cli/HISTORY.rst [hunk 12]
```diff
@@ -3,6 +3,119 @@
...
+* Add --uptime-sla (#12772)
+* Update containerservice package (#12964)
+* Add node public IP support in CLI extension. (#13015)
```
Comment (qianwens) [inline src/azure-cli/HISTORY.rst:19]:
```suggestion
* Add node public IP support. (#13015)
```

File: src/azure-cli/HISTORY.rst [hunk 13]
```diff
@@ -3,6 +3,119 @@
...
+**Monitor**
+
+* Support private link scenario for monitor (#12931)
+
+**Network**
+
+* Deprecate sku for public ip update command (#12898)
+* `az network private-endpoint`: Support private dns zone group. (#13038)
+* Enable local context feature for vnet/subnet parameter (#13059)
+
```
Comment (haroldrandom) [inline src/azure-cli/HISTORY.rst:81]:
```suggestion
* fix wrong usage example in test_nw_flow_log_delete

```

File: src/azure-cli/HISTORY.rst [hunk 14]
```diff
@@ -3,6 +3,119 @@
...
+* Fix #12891: az sf application update --application-parameters removes old parameters that are not in the request (#12992)
+
+**ServiceFabric**
+
+* Fix #12470 az sf create cluster, fix bugs in update durability and reliability and find vmss correctly through the code given a node type name (#12731)
+
+**SQL**
+
+* Add az sql mi op list, az sql mi op get, az sql mi op cancel (#12667)
```
Comment (qianwens) [inline src/azure-cli/HISTORY.rst:104]:
```suggestion
* Add `az sql mi op list`, `az sql mi op get`, `az sql mi op cancel` (#12667)
```

File: src/azure-cli/HISTORY.rst [hunk 15]
```diff
@@ -3,6 +3,119 @@
...
+
+* `az ad app create/update`: support --optional-claims as a parameter (#12954)
+
+**RDBMS**
+
+* Add Azure active directory administrator commands for PostgreSQL and MySQL (#12812)
+
+**Service Fabric**
+
+* Fix #12891: az sf application update --application-parameters removes old parameters that are not in the request (#12992)
```
Comment (qianwens) [inline src/azure-cli/HISTORY.rst:96]:
```suggestion
* Fix #12891: `az sf application update --application-parameters` removes old parameters that are not in the request (#12992)
```

File: src/azure-cli/HISTORY.rst [hunk 16]
```diff
@@ -3,6 +3,119 @@
...
+
+**ServiceFabric**
+
+* Fix #12470 az sf create cluster, fix bugs in update durability and reliability and find vmss correctly through the code given a node type name (#12731)
+
+**SQL**
+
+* Add az sql mi op list, az sql mi op get, az sql mi op cancel (#12667)
+* Sql midb Add: update/show long term retention policy,  show/delete long term retention backups, restore long term retention backup (#12712)
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:105]:
```suggestion
* `az sql midb`: update/show long term retention policy,  show/delete long term retention backups, restore long term retention backup (#12712)
```

File: src/azure-cli/HISTORY.rst [hunk 17]
```diff
@@ -3,6 +3,119 @@
...
+* Update Batch API (#12813)
+
+**Compute**
+
+* `az sig image-version create`: Add storage account type Premium_LRS (#12919)
+* `az vmss update`: Fix terminate notification update issue (#12948)
+* `az vm/vmss create`: Add support for specialized image version (#12997)
+* SIG API Version 2019-12-01. (#12899)
+* `az sig image-version create`: Add --target-region-encryption. (#12899)
+
```
Comment (haroldrandom) [inline src/azure-cli/HISTORY.rst:62]:
```suggestion
* Fix tests fail when running in serial due to keyvault name is duplicated in global in-momery cache

```

File: src/azure-cli-core/HISTORY.rst [hunk 18]
```diff
@@ -3,6 +3,14 @@
 Release History
 ===============
 
+2.4.0
+++++++
+* Upgrade msrestazure to 0.6.3 (#12839)
+* Output more error details of exceptions if possible (#12945)
+* Support track1 and track2 mgmt SDK side by side in CLI (#12952)
+* Fix generic remove property NoneType exception (#12950)
+* `az --version`: Show command instruction and detailed instruction link when updates available (#12981)
```
Comment (qianwens) [inline src/azure-cli-core/HISTORY.rst:12]:
```suggestion
* `az --version`: Show command instruction and detailed instruction link when updates available (#12981)
* Fix bug in core test test_parser that will affect exception stack while existing if argument parse raise Exception
* Make CLI respect file logging configuration option: enable_log_file and AZURE_LOGGING_ENABLE_LOG_FILE env variable
* Fix the name of extension of dev type is None if it has incompatibility problem with core
```

File: src/azure-cli/HISTORY.rst [hunk 19]
```diff
@@ -3,6 +3,116 @@
...
+* Support private link scenario for monitor (#12931)
+* fix wrong mocking way in test_monitor_general_operations.py
+
+**Network**
+
+* Deprecate sku for public ip update command (#12898)
+* `az network private-endpoint`: Support private dns zone group. (#13038)
+* Enable local context feature for vnet/subnet parameter (#13059)
+* fix wrong usage example in test_nw_flow_log_delete
```
Comment (qianwens) [inline src/azure-cli/HISTORY.rst:80]:
```suggestion
* Fix wrong usage example in test_nw_flow_log_delete
```

File: src/azure-cli/HISTORY.rst [hunk 20]
```diff
@@ -3,6 +3,116 @@
...
+
+**IoT Central**
+
+* Deprecate `az iotcentral` (#12681)
+* Add `az iot central` command module (#12681)
+
+**Monitor**
+
+* Support private link scenario for monitor (#12931)
+* fix wrong mocking way in test_monitor_general_operations.py
```
Comment (qianwens) [inline src/azure-cli/HISTORY.rst:73]:
```suggestion
* Fix wrong mocking way in test_monitor_general_operations.py
```

File: src/azure-cli/HISTORY.rst [hunk 21]
```diff
@@ -3,6 +3,116 @@
...
+* `az sig image-version create`: Add storage account type Premium_LRS (#12919)
+* `az vmss update`: Fix terminate notification update issue (#12948)
+* `az vm/vmss create`: Add support for specialized image version (#12997)
+* SIG API Version 2019-12-01. (#12899)
+* `az sig image-version create`: Add --target-region-encryption. (#12899)
+* Fix tests fail when running in serial due to keyvault name is duplicated in global in-momery cache
+
+**CosmosDB**
+
+* Support az cosmosdb private-link-resource and private-endpoint-connection (#12960)
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:63]:
```suggestion
* Support `az cosmosdb private-link-resource/private-endpoint-connection` (#12960)
```

File: src/azure-cli/HISTORY.rst [hunk 22]
```diff
@@ -3,6 +3,116 @@
 Release History
 ===============
 
+2.4.0
+++++++
+
+**ACR**
+
+* `az acr run --cmd`: disable working directory override (#12877)
+* Dedicated data endpoint (#12967)
```
Comment (qianwens) [inline src/azure-cli/HISTORY.rst:12]:
```suggestion
* Support dedicated data endpoint (#12967)
```

**Pull request merged**